docs/content/technical/build_process.md

```diff
@@ -0,0 +1,115 @@
+---
+title: 'Build process information'
+weight: 35
+---
+
+## 1. Introduction
+
+NDS ROMs are complicated. They have an ARM9 binary, an ARM7 binary, an optional
+filesystem image, and a ROM header with information about where to locate all
+the parts of the ROM, as well as other information like the game title and icon.
+
+You can get more information about it [in GBATEK](https://problemkaputt.de/gbatek.htm#dscartridgeheader).
+
+This document explains the way BlocksDS builds each component and assembles the
+final ROM.
+
+## 2. ARM7 and ARM9 binaries
+
+This part is pretty standard. The developer provides source code files that
+are compiled and linked into an `.elf` file (one per CPU). The source code files
+may contain code or data as requested by the developer. This code and data is
+placed into different sections as with any other `.elf` file:
+
+- `text`: Code.
+- `rodata`: Constant data.
+- `data`: Initialized variables.
+- `bss`: Uninitialized variables (they start as 0).
+
+This `elf` file is also very useful. It can be used with binutils, for example,
+to analyse which functions are the biggest ones in your program (with
+`nm --size-sort file.elf`). It can also be used to debug games in emulators that
+support `gdb`, like melonDS and DesMuMe.
+
+The DSi complicates all of this a bit more. The DSi has 16 MiB of RAM, while a
+regular DS only has 4 MiB. It's very useful for any DSi-compatible game to be
+able to not load DSi-related code and data when running on a regular DS.  This
+is done by defining other sections that are ignored in a regular DS, and they
+are loaded to RAM in a DSi:
+
+- `twl`: Combined `text`, `rodata` and `data` for DSi.
+- `twl_bss`: Uninitialized DSi-only variables.
+
+The way the `twl` sections are used is explained below, in the section about
+creation of NDS ROMs.
+
+Note that building an ARM7/ARM9 binary for the CPU can't be done with a generic
+toolchain without any modifications. It is needed to provide a linkerscript that
+understands the memory layout of the DS. You also need a special `crt0` that can
+set the hardware to a sensible state before `main()` is called. For example, the
+MPU (Memory Protection Unit) of the ARM9 needs to be setup correctly, and it is
+needed to ensure that things like timers, DMA, etc, are set to an idle state.
+Unfortunately, this also means that the `crt0` depends on `libnds`, which is the
+one that contains the hardware initialization code.
+
+All of this means that invoking `gcc` isn't as simple as running `gcc file.c`.
+It also means that `gcc` updates may require small changes to the command line
+options used. Some of this is hidden from the user in `specs` files located
+[here](https://github.com/blocksds/sdk/tree/master/sys/crts). This is done
+purely for convenience. You can skip the `specs` files and copy the build
```

Comment:
for end user convenience and to reduce the amount of breaking changes to user project Makefiles